From 5b4a3596a712b9a0d0ed8e2bb972bbe7eaaae848 Mon Sep 17 00:00:00 2001
From: Peter Griffin <peter.griffin@linaro.org>
Date: Thu, 12 Mar 2020 12:57:06 +0100
Subject: [PATCH 10/17] MediaSystem: output CdmStatus string on error of
 generateRequest()

Make use of the new helper function to output CdmStatus error
string in case of error with generateRequest().

Signed-off-by: Peter Griffin <peter.griffin@linaro.org>
Change-Id: I00d1d2606c4c2e001b202fa6297b16fc2f5ddb07
---
 CMakeLists.txt   | 4 ++--
 MediaSession.cpp | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9c81a69..0e29d6c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -24,8 +24,8 @@ set(PLUGIN_SOURCES
 add_library(${PLUGIN_NAME} SHARED ${PLUGIN_SOURCES})
 target_compile_definitions(${PLUGIN_NAME} PRIVATE ${WIDEVINE_FLAGS})
 target_include_directories(${PLUGIN_NAME} PRIVATE ${PLUGINS_INCLUDE_DIR} ${WIDEVINE_INCLUDE_DIRS})
-target_link_libraries(${PLUGIN_NAME} PRIVATE ${PLUGIN_LIBS} ${NAMESPACE}Core::${NAMESPACE}Core ${WIDEVINE_LIBRARIES})
+target_link_libraries(${PLUGIN_NAME} PRIVATE ${PLUGIN_LIBS} ${NAMESPACE}Core::${NAMESPACE}Core ${WIDEVINE_LIBRARIES} ssl)
 set_target_properties(${PLUGIN_NAME} PROPERTIES SUFFIX ".drm")
 set_target_properties(${PLUGIN_NAME} PROPERTIES PREFIX "")
 
-install(TARGETS ${PLUGIN_NAME} DESTINATION ${CMAKE_INSTALL_PREFIX}/share/${NAMESPACE}/OCDM)
\ No newline at end of file
+install(TARGETS ${PLUGIN_NAME} DESTINATION ${CMAKE_INSTALL_PREFIX}/share/${NAMESPACE}/OCDM)
diff --git a/MediaSession.cpp b/MediaSession.cpp
index e553bbf..4c78022 100644
--- a/MediaSession.cpp
+++ b/MediaSession.cpp
@@ -105,8 +105,8 @@ void MediaKeySession::Run(const IMediaKeySessionCallback *f_piMediaKeySessionCal
 
     widevine::Cdm::Status status = m_cdm->generateRequest(m_sessionId, m_initDataType, m_initData);
     if (widevine::Cdm::kSuccess != status) {
-       TRACE_L1("%s: generateRequest() failed\n", __FUNCTION__);
-       m_piCallback->OnKeyMessage((const uint8_t *) "", 0, "");
+      TRACE_L1("%s: generateRequest() failed CdmStatus=%s", __FUNCTION__, widevineStatusToCString(status));
+      m_piCallback->OnKeyMessage((const uint8_t *) "", 0, "");
     }
   }
   else {
-- 
2.17.1

